raiden/tests/unit/migrations/test_v18_to_v19.py

```diff
@@ -0,0 +1,102 @@
+import json
+from datetime import datetime
+from pathlib import Path
+from unittest.mock import patch
+
+from raiden.storage.sqlite import SQLiteStorage
+from raiden.tests.utils.migrations import create_fake_web3_for_block_hash
+from raiden.utils.upgrades import UpgradeManager
+
+
+def setup_storage(db_path):
+    storage = SQLiteStorage(str(db_path))
+
+    # Add the v18 state changes to the DB
+    state_changes_file = Path(__file__).parent / 'data/v18_statechanges.json'
+    state_changes_data = json.loads(state_changes_file.read_text())
+    for state_change_record in state_changes_data:
+        storage.write_state_change(
+            state_change=json.dumps(state_change_record[1]),
+            log_time=datetime.utcnow().isoformat(timespec='milliseconds'),
+        )
+
+    # Add the v18 events to the DB
+    events_file = Path(__file__).parent / 'data/v18_events.json'
+    events_data = json.loads(events_file.read_text())
+    for event in events_data:
+        state_change_identifier = event[1]
+        event_data = json.dumps(event[2])
+        log_time = datetime.utcnow().isoformat(timespec='milliseconds')
+        event_tuple = (
+            None,
+            state_change_identifier,
+            log_time,
+            event_data,
+        )
+        storage.write_events(
+            state_change_identifier=state_change_identifier,
+            events=[event_tuple],
+            log_time=log_time,
+        )
+
+    # Finally add a chain state. Using the v17 already existing chainstate here out of laziness
+    # and since for the purpose of the attributes of the chain state we migrate in v18->v19 it
+    # should not mattter
+    chain_state_data = Path(__file__).parent / 'data/v17_chainstate.json'
```

Comment:
I don't get why you wouldn't be testing the migration on v18 chain state if 
> it should not matter

Reply:
Because as I said in the comment [above](https://github.com/raiden-network/raiden/pull/3550#discussion_r261097826) it's extra work to get it and since it's identical for the purposes of the test there is no reason to do it.

Reply:
I made a new v18 chainstate to add some pending transacitons so this is no longer the correct diff.